File: C/Convert-BST-to-Greater-Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

void compute(struct TreeNode* root, int* sum) {
    if(!root) return;
    compute(root->right, sum);
    *sum += root->val;
    root->val = *sum;
    compute(root->left, sum);
}

struct TreeNode* convertBST(struct TreeNode* root){
    int sum = 0;
           
    compute(root, &sum);
        
    return root;
}

File: Python3/convert_BST_to_greater_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        self.sum = 0
        
        def compute(root):
            if not root: return
            compute(root.right)
            self.sum += root.val
            root.val = self.sum
            compute(root.left)
        compute(root)
        
        return root
